test/functional/autocmd/signal_spec.lua

```diff
@@ -44,3 +47,64 @@ describe('autocmd Signal', function()
     eq(nil, next_msg(500))
   end)
 end)
+
+describe('autowriteall on kill', function()
+  before_each(clear)
+
+  local function test_deadly_sig(signame, awa, should_write)
+    local testfile = 'Xtest_SIG' .. signame .. (awa and '_awa' or '_noawa')
+    local teststr = 'Testaaaaaaa'
+
+    if awa then
+      command('set awa')
+    end
+
+    command('edit ' .. testfile)
+    feed('i' .. teststr)
+    posix_kill(signame, fn.getpid())
```

Comment:
`vim.uv.kill()` should work, can you try that instead? Then we can drop `posix_kill`. And then some of these tests can run on Windows. I don't want to bother you with that for the old tests in this file, but at least for SIGHUP, and possibly SIGTERM in your new tests? https://docs.libuv.org/en/v1.x/signal.html

To be clear:

- move your new tests above the `skip(is_os('win')` line 15
- run your tests for at least a couple supported signals on Windows.